sync/http/errors.go

```diff
@@ -0,0 +1,60 @@
+package http
+
+import (
+	gohttp "net/http"
+)
+
+// CustomError defines an abstract struct that can represent several types of HTTP errors.
```

Comment:
The error can be plain `type error struct{}` no need to export it and name it custom.

Reply:
Turns out the error type has to be exported because the `New**Error` exported functions in `errors.go` all return this type to their caller and the Go linter complains about `exported functions returning non-exported value, which can be annoying to use`. Given that, I think the best option is to rename the error struct to `type Error`.